### PR TITLE
Edit External Cluster KubeOne CRD 

### PR DESCRIPTION
**Pull request by imharshita**
Signed-off-by: Harshita sharma <harshita.sharma6174@gmail.com>

**What does this PR do / Why do we need it**: Edit External Cluster KubeOne CRD and renaming fields

**Does this PR close any issues?**:
Fixes #

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: pkg/apis/kubermatic/v1/external_cluster.go [hunk 1]
```diff
@@ -74,8 +74,42 @@ type ExternalClusterCloudSpec struct {
...
+)
+
+// ExternalClusterStatus defines the kubeone external cluster status.
+type KubeOneExternalClusterStatus struct {
+	State         KubeOneExternalClusterState `json:"state"`
+	StatusMessage string                      `json:"statusMessage,omitempty"`
+}
+
 type ExternalClusterKubeOneCloudSpec struct {
-	Name                 string                                 `json:"name"`
```
Comment (imharshita) [inline pkg/apis/kubermatic/v1/external_cluster.go:78]: Removed Name field as we already have name stored here: https://github.com/kubermatic/kubermatic/blob/51984b32b361d326b82af52fcd45d5eac986cd48/pkg/apis/kubermatic/v1/external_cluster.go#L63

File: pkg/provider/kubernetes/external_cluster.go [hunk 2]
```diff
@@ -511,11 +511,3 @@ func (p *ExternalClusterProvider) CreateOrUpdateKubeOneSSHSecret(ctx context.Con
 
 	return nil
 }
-
```
Comment (imharshita) [inline pkg/provider/kubernetes/external_cluster.go:514]: removed these methods as kubeone resources will be created in a dedicated namespace named "kubeone-`cluster-id`" so providing cluster name in secret name is not required anymore

File: pkg/provider/kubernetes/credentials.go [hunk 3]
```diff
@@ -995,33 +1005,33 @@ func createOrUpdateKubeOneNutanixSecret(ctx context.Context, cloud apiv2.KubeOne
 	return nil
 }
 
-func GetKubeOneCredentialsSecretName(cloud apiv2.KubeOneCloudSpec, externalCluster *kubermaticv1.ExternalCluster) string {
+func GetKubeOneCredentialsSecretName(cloud apiv2.KubeOneCloudSpec) string {
 	if cloud.AWS != nil {
```
Comment (imharshita) [inline pkg/provider/kubernetes/credentials.go:1009]: removed these methods as kubeone resources will be created in a dedicated namespace named "kubeone-`cluster-id`"so providing cluster name in secret name is not required anymore

File: pkg/apis/kubermatic/v1/external_cluster.go [hunk 4]
```diff
@@ -74,8 +74,42 @@ type ExternalClusterCloudSpec struct {
 	KubeOne *ExternalClusterKubeOneCloudSpec `json:"kubeone,omitempty"`
 }
 
+type KubeOneExternalClusterState string
+
+const (
+	// PROVISIONING state indicates the cluster is being imported.
+	PROVISIONING KubeOneExternalClusterState = "PROVISIONING"
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/external_cluster.go]: Please use regular constant names like "Provisioning". We might even want some suffix or prefix to make this set of constant more distinct, like `ProvisioningStatus` or `StatusProvisioning` maybe.

File: pkg/resources/resources.go [hunk 5]
```diff
@@ -544,14 +544,27 @@ const (
...
+	KubeOneManifest           = "manifest"
+	KubeOneAWS                = "AWS"
+	KubeOneGCP                = "GCP"
+	KubeOneAzure              = "Azure"
+	KubeOneDigitalOcean       = "DigitalOcean"
+	KubeOneHetzner            = "Hetzner"
+	KubeOneNutanix            = "Nutanix"
+	KubeOneOpenStack          = "OpenStack"
+	KubeOneEquinix            = "Equinix"
+	KubeOneVSphere            = "VSphere"
```
Comment (xrstf) [inline pkg/resources/resources.go]: Are we sure that those constant values should be uppercased?

File: pkg/resources/resources.go [hunk 6]
```diff
@@ -583,21 +597,23 @@ const (
...
 
-	NutanixUsername      = "username"
-	NutanixPassword      = "password"
-	NutanixCSIUsername   = "csiUsername"
-	NutanixCSIPassword   = "csiPassword"
+	NutanixUsername    = "username"
+	NutanixPassword    = "password"
+	NutanixCSIUsername = "csiUsername"
+	NutanixCSIPassword = "csiPassword"
+	NutanixProxyURL    = "proxyURL"
```
Comment (xrstf) [inline pkg/resources/resources.go:614]: What has this change to do with this PR?

File: pkg/apis/kubermatic/v1/external_cluster.go [hunk 7]
```diff
@@ -74,8 +74,42 @@ type ExternalClusterCloudSpec struct {
...
+
+	// ERROR state indicates the cluster is unusable. It will be automatically deleted. Details can be found in the
+	// `statusMessage` field.
+	ERROR KubeOneExternalClusterState = "ERROR"
+)
+
+// ExternalClusterStatus defines the kubeone external cluster status.
+type KubeOneExternalClusterStatus struct {
+	State         KubeOneExternalClusterState `json:"state"`
+	StatusMessage string                      `json:"statusMessage,omitempty"`
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/external_cluster.go]: Can we please not mix "state" and "status"? :)

File: pkg/apis/kubermatic/v1/external_cluster.go (hunk 4 above)
Comment (xrstf) [reply pkg/apis/kubermatic/v1/external_cluster.go]: Also, where are these constants coming from? Are we defining them ourselves? If so, can we make them non-screaming? I presume that we want to show them when admins to a `kubectl get externalclusters` and HAVING THE LIST SCREAM AT THE USER IS SUPER RUDE AND ALSO UPPERCASE IS HARD TO READ AT LEAST TO ME.

Comment (xrstf) [reply pkg/apis/kubermatic/v1/external_cluster.go]: For comparision, our project phases are also "Active", "Inactive" and "Terminating", not uppercased. :)

Comment (imharshita) [reply pkg/apis/kubermatic/v1/external_cluster.go]: agreed

**Pull request merged**